kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```
@@ -2552,45 +2494,57 @@ public boolean allowRemoveCollection(String id)
 	 */
 	protected boolean allowRemove(String id)
 	{
-		
+
+		// check security to delete any collection
+		if ( unlockCheck(AUTH_RESOURCE_REMOVE_ANY, id) )
+			return true;
+
+		// check security to delete own collection
+		else
+			return allowOwner(id, AUTH_RESOURCE_REMOVE_OWN);
+
+	} // allowRemove
+
+	/**
+	 * This checks to see if some permission checks should pass because the current
+	 * user is the creator
+	 * @param id
+	 * @param permission
+	 * @return
+	 */
+	private boolean allowOwner(String id, String permission) {
+
 		String currentUser = sessionManager.getCurrentSessionUserId();
 		String owner = "";
-		
-		//Supper users always have the permission
-		if (m_securityService.isSuperUser()) {
-			return true;
-		}
-		
+
 		try
 		{
 			ResourceProperties props = getProperties(id);
 			owner = props.getProperty(ResourceProperties.PROP_CREATOR);
 		}
-		catch ( Exception e ) 
+		catch ( PermissionException e )
 		{
 			// PermissionException can be thrown if not RESOURCE_AUTH_READ
 			return false;
 		}
```

Is there any other way of doing this?  Empty catch blocks to control program flow...not ideal practice.  Bit of a rewrite probably?  That said, there are lots of examples of this so...